### PR TITLE
http_test:wait until http is already started before http test.

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,all,
   This PR is used to fix bug in http test.  If we do not wait until http is already started, we may meet the following failure in http test:
```
2017/03/18 11:55:38 http_status.go:78: [0;37m[info] Listening on :10090 for status and metrics report.[0m

----------------------------------------------------------------------
FAIL: region_handler_test.go:159: TidbRegionHandlerTestSuite.TestGetRegionByIDWithError

region_handler_test.go:163:
    c.Assert(err, IsNil)
... value *url.Error = &url.Error{Op:"Get", URL:"http://127.0.0.1:10090/regions/xxx", Err:(*net.OpError)(0xc420624910)} ("Get http://127.0.0.1:10090/regions/xxx: dial tcp 127.0.0.1:10090: getsockopt: connection refused")
```

@coocood @zimulala @disksing PTAL.

### CODE REVIEW

File: server/server_test.go [hunk 1]
```diff
@@ -603,13 +604,33 @@ func getStmtCnt(content string) (stmtCnt map[string]int) {
...
 	}
+	if retry == retryTime {
+		log.Fatalf("Failed to connect db for %d retries in every %v ms", retryTime, time.Millisecond*10)
+	}
+	// connect http status
+	statusURL := fmt.Sprintf("http://127.0.0.1%s/status", statusAddr)
+	for retry = 0; retry < retryTime; retry++ {
+		resp, err := http.Get(statusURL)
+		if err == nil {
+			resp.Body.Close()
```
Comment (siddontang) [inline server/server_test.go]: for get, please read all data from body before close.

File: server/server_test.go [hunk 2]
```diff
@@ -603,13 +604,34 @@ func getStmtCnt(content string) (stmtCnt map[string]int) {
...
+	for ; retry < retryTime; retry++ {
 		time.Sleep(time.Millisecond * 10)
 		db, err := sql.Open("mysql", dsn)
 		if err == nil {
 			db.Close()
 			break
 		}
 	}
+	if retry == retryTime {
+		log.Fatalf("Failed to connect db for %d retries in every %v ms", retryTime, time.Millisecond*10)
```
Comment (disksing) [inline server/server_test.go]: The log will be '10ms ms'...

**Comment by disksing** [comment]
LGTM. PTAL @shenli

**Comment by shenli** [comment]
LGTM

**Pull request merged**